app/models/reports/schema_v2.rb

```diff
@@ -141,23 +95,11 @@ def registered_patients_query_v2(region)
     end
 
     memoize def controlled
-      regions.each_with_object({}).each do |region, hsh|
-        if earliest_patient_recorded_at[region.slug].nil?
-          hsh[region.slug] = Hash.new(0)
-          next
-        end
-        hsh[region.slug] = control_rate_query_v2.controlled_counts(region, range: active_range(region))
-      end
+      regions.each_with_object({}) { |region, hsh| hsh[region.slug] = sum(region, :controlled_under_care) }
     end
 
     memoize def uncontrolled
-      regions.each_with_object({}).each do |region, hsh|
-        if earliest_patient_recorded_at[region.slug].nil?
-          hsh[region.slug] = Hash.new(0)
-          next
-        end
-        hsh[region.slug] = control_rate_query_v2.uncontrolled_counts(region, range: active_range(region))
-      end
+      regions.each_with_object({}) { |region, hsh| hsh[region.slug] = sum(region, :uncontrolled_under_care) }
```

Comment:
These came over nicely to the new single-query code, and the old PatientState query can be removed.

Reply:
🙌  sweet!